Mathlib/Data/Nat/Fib/Lemmas.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2025 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import Mathlib.Data.Nat.Fib.Basic
+import Mathlib.LinearAlgebra.Matrix.Notation
+import Mathlib.LinearAlgebra.Matrix.Determinant.Basic
+
+/-!
+# Cassini and Catalan identities for the Fibonacci numbers
+
+Cassini's identity states that for nonzero `n`, `fib (n + 1) * fib (n - 1) - fib n ^ 2` is equal
+to `(-1) ^ n`. And Catalan's identity states that for nonzero `x` and `a`, we get
+`fib (x + a) ^ 2 - fib x * fib (x + 2 * a) = (-1) ^ x * fib a ^ 2`.
+-/
+
+namespace Nat
+
+/-- For any nonzero `n`, the matrix `[fib (n + 1), fib n; fib n, fib (n - 1)]` is equal to
+`[1, 1; 1, 0] ^ n`. -/
```

Comment:
I'm not a fan of docstrings that just repeat the theorem statements verbatim. I'd personally write something like

```suggestion
/-- Being a linear recurrence, the entries of the Fibonacci sequence can be related to matrix exponentiation. -/
```
But you're free to change up the words.